src/Fabulous/ViewElement.fs

```diff
@@ -146,6 +152,13 @@ type ViewElement internal (targetType: Type, create: (unit -> obj), update: (Vie
         match attribs |> Array.tryFind (fun kvp -> kvp.Key = key.KeyValue) with 
         | Some kvp -> unbox<'T>(kvp.Value)
         | None -> failwithf "Property '%s' does not exist on %s" key.Name x.TargetType.Name
+        
+    member x.TryGetKey() =
+        match x.TryGetAttributeKeyed(ViewElement._KeyAttribKey) with
+        | ValueSome key -> Some key
+        | ValueNone -> None
```

Comment:
Please use `ValueOption` instead of `Option` when working on the internal plumbing of Fabulous.
ValueOption is a struct and is lighter on the memory (stored on Stack instead of the GC-managed Heap).
So I think you can directly return the result of TryGetAttributeKeyed